File: ZemljaSlova.Model/Reports/MembersReport.cs
using System;
using System.Collections.Generic;

namespace ZemljaSlova.Model.Reports
{
    public class MembersReport
    {
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public int TotalActiveMembers { get; set; }
        public int NewMembersInPeriod { get; set; }
        public int ExpiredMemberships { get; set; }
        public int TotalMemberships { get; set; }
        public string ReportPeriod { get; set; } = string.Empty;
        public List<MemberSummary> MemberSummaries { get; set; } = new List<MemberSummary>();
        public List<MembershipActivity> MembershipActivities { get; set; } = new List<MembershipActivity>();
    }

    public class MemberSummary
    {
        public int MemberId { get; set; }
        public string MemberName { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public DateTime? MembershipStartDate { get; set; }
        public DateTime? MembershipEndDate { get; set; }
        public bool IsActive { get; set; }
        public int TotalRentals { get; set; }
        public int TotalPurchases { get; set; }
    }

    public class MembershipActivity
    {
        public int Id { get; set; }
        public string MemberName { get; set; } = string.Empty;
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public string Status { get; set; } = string.Empty;
    }
}

File: UI/zemljaslova_desktop/lib/models/reports/members_report.dart
import 'package:json_annotation/json_annotation.dart';

part 'members_report.g.dart';

@JsonSerializable()
class MembersReport {
  @JsonKey(name: 'startDate')
  final DateTime startDate;
  
  @JsonKey(name: 'endDate')
  final DateTime endDate;
  
  @JsonKey(name: 'totalActiveMembers')
  final int totalActiveMembers;
  
  @JsonKey(name: 'newMembersInPeriod')
  final int newMembersInPeriod;
  
  @JsonKey(name: 'expiredMemberships')
  final int expiredMemberships;
  
  @JsonKey(name: 'totalMemberships')
  final int totalMemberships;
  
  @JsonKey(name: 'reportPeriod')
  final String reportPeriod;
  
  @JsonKey(name: 'memberSummaries')
  final List<MemberSummary> memberSummaries;
  
  @JsonKey(name: 'membershipActivities')
  final List<MembershipActivity> membershipActivities;

  MembersReport({
    required this.startDate,
    required this.endDate,
    required this.totalActiveMembers,
    required this.newMembersInPeriod,
    required this.expiredMemberships,
    required this.totalMemberships,
    required this.reportPeriod,
    required this.memberSummaries,
    required this.membershipActivities,
  });

  factory MembersReport.fromJson(Map<String, dynamic> json) => _$MembersReportFromJson(json);
  Map<String, dynamic> toJson() => _$MembersReportToJson(this);
}

@JsonSerializable()
class MemberSummary {
  @JsonKey(name: 'memberId')
  final int memberId;
  
  @JsonKey(name: 'memberName')
  final String memberName;
  
  @JsonKey(name: 'email')
  final String email;
  
  @JsonKey(name: 'membershipStartDate')
  final DateTime? membershipStartDate;
  
  @JsonKey(name: 'membershipEndDate')
  final DateTime? membershipEndDate;
  
  @JsonKey(name: 'isActive')
  final bool isActive;
  
  @JsonKey(name: 'totalRentals')
  final int totalRentals;
  
  @JsonKey(name: 'totalPurchases')
  final int totalPurchases;

  MemberSummary({
    required this.memberId,
    required this.memberName,
    required this.email,
    this.membershipStartDate,
    this.membershipEndDate,
    required this.isActive,
    required this.totalRentals,
    required this.totalPurchases,
  });

  factory MemberSummary.fromJson(Map<String, dynamic> json) => _$MemberSummaryFromJson(json);
  Map<String, dynamic> toJson() => _$MemberSummaryToJson(this);
}

@JsonSerializable()
class MembershipActivity {
  final int id;
  
  @JsonKey(name: 'memberName')
  final String memberName;
  
  @JsonKey(name: 'startDate')
  final DateTime startDate;
  
  @JsonKey(name: 'endDate')
  final DateTime endDate;
  
  @JsonKey(name: 'status')
  final String status;

  MembershipActivity({
    required this.id,
    required this.memberName,
    required this.startDate,
    required this.endDate,
    required this.status,
  });

  factory MembershipActivity.fromJson(Map<String, dynamic> json) => _$MembershipActivityFromJson(json);
  Map<String, dynamic> toJson() => _$MembershipActivityToJson(this);
}
